source/common/runtime/runtime_impl.cc

```diff
@@ -231,6 +231,10 @@ DiskLayer::DiskLayer(const std::string& name, const std::string& path,
 
 void DiskLayer::walkDirectory(const std::string& path, const std::string& prefix) {
   ENVOY_LOG(debug, "walking directory: {}", path);
+  // Check if this is an obviously bad path.
```

Comment:
Per offline convo, I still think the correct solution here is a max recursion check. Can we TODO that? Additionally, It seems like we should potentially do this check in any place we attempt to load a file (TLS certs, etc.)? Should we issue/TODO that also?

Reply:
Agree. Added TODO for recursion depth and filed https://github.com/envoyproxy/envoy/issues/3170.